habitat/sims/habitat_simulator/habitat_simulator.py

```diff
@@ -283,6 +333,21 @@ def _update_agents_state(self) -> bool:
 
         return is_updated
 
+    def _load_agent_body_meshes(self) -> None:
```

Comment:
We really need to think about this when we start managing object states in episodes. @jturner65 Is there not a cleaner way to load the agent_body_mesh_config? I don't want to load it multiple times if its unnecessary.

Reply:
This could certainly be made quite a bit simpler - all of this could be bundled in simulator.cpp on sim side and could be accessed by a single function call; probably would be useful to have agent configs be supported in the scene instance json.

Reply:
I really disagree with the later point. Agent configs definitely belong as part of the task definition not the scene config.

Reply:
Fair enough.  At the very least, the attributes for the objects being used to represent the various agents could be included in the Scene Dataset Library configs.  That way the agent configuration object attributes would not need to be specifically loaded when assigning an object to the agent.